std/range/package.d

```diff
@@ -6674,16 +6676,37 @@ private:
     ]));
 }
 
+// Issue 17742
+@safe unittest
+{
+    import std.algorithm.iteration : map;
+    import std.algorithm.comparison : equal;
+    auto ror = 5.iota.map!(y => 5.iota.map!(x => x * y).array).array;
+    assert(ror[3][2] == 6);
+    auto result = transposed!(TransverseOptions.assumeNotJagged)(ror);
+    assert(result[2][3] == 6);
+
+    auto x = [[1,2,3],[4,5,6]];
+    auto y = transposed!(TransverseOptions.assumeNotJagged)(x);
+    assert(y.front.equal([1,4]));
+    assert(y[0].equal([1,4]));
+    assert(y[0][0] == 1);
+    assert(y[1].equal([2,5]));
+    assert(y[1][1] == 5);
```

Comment:
Some more tests to ensure we get this right:

```D
import std.range.interfaces;
// limit input range to only a forward range
ForwardRange fr = inputRangeObject(x);

auto z = transposed!(TransverseOptions.assumeNotJagged)(fr);
assert(z.front.equal([1, 4]));
assert(z[0].equal([1,4]));
assert(z[1].equal([2, 5]));
assert(!is(typeof(z[0][0]))); // cannot index a forward range
```

Reply:
And once you implement the full gamut of `TransverseOptions`, those should be tested.

Reply:
After spending more time with this I realized there is a problem, this fails:
```
static assert(hasAssignableElements!(typeof(fr)));
// aka
auto z = transposed!.... // will fail
```

Reply:
Ugh ok. We will have to make a custom harness to test. Nothing’s ever easy ;)

Reply:
Oops never mind I see you found an easy way!